modules/channel-smooch/src/backend/index.ts

```diff
@@ -0,0 +1,75 @@
+import * as sdk from 'botpress/sdk'
+
+import { Config } from '../config'
+
+import { removeMiddleware, setupMiddleware, SmoochClient } from './client'
+import { Clients } from './typings'
+
+let router
+const route = '/webhook'
+const clients: Clients = {}
+
+const onServerStarted = async (bp: typeof sdk) => {
+  await setupMiddleware(bp, clients)
+}
+
+const onServerReady = async (bp: typeof sdk) => {
+  router = bp.http.createRouterForBot('channel-smooch', {
+    checkAuthentication: false
+  })
+
+  router.post(route, async (req, res) => {
+    const { botId } = req.params
+    const client = clients[botId]
+
+    if (client && client.auth(req)) {
+      await client.handleWebhookRequest(req.body)
+      res.sendStatus(200)
+    } else {
+      res.status(404).send('Bot not a smooch bot')
+    }
```

Comment:
The message can be misleading, if the secret is invalid, it will say "Bot not a smooch bot". I would have a different message for bot id invalid / authentication token invalid